### PR TITLE
[Miner] chore: scaffold create-claim message

### PR DESCRIPTION
**Pull request by bryanchriswhite**
## Summary

This message type is a nested dependency of the `Miner` via the `SupplierClient`.

- Scaffolded `MsgCreateClaim` message types, handlers, etc.:
  ```
  ignite scaffold message create-claim session_header:SessionHeader root_hash --signer SupplierAddress --module supplier
  ```
  (NB: generated a temporary SessionHeader duplicate in the supplier module so the above would work; subsequently, removed and updated protobuf import and usage.)
- Switch `MsgCreateClaim#root_hash` type from `string` to `[]byte`  & refactor usages.

## Issue

Related to:
- #13 
- #7 

This message type is a nested dependency of the `Miner` via the `SupplierClient`.

## Type of change

Select one or more:

- [x] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

- [x] **Run all unit tests**: `make go_test`
- [x] **Verify Localnet manually**: See the instructions [here](TODO: add link to instructions)

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [x] I have performed a self-review of my own code
- [x] I have commented my code, updated documentation and left TODOs throughout the codebase

### CODE REVIEW

File: proto/pocket/supplier/tx.proto [hunk 1]
```diff
@@ -1,7 +1,18 @@
...
 // Msg defines the Msg service.
-service Msg {}
\ No newline at end of file
+service Msg {
+  rpc CreateClaim (MsgCreateClaim) returns (MsgCreateClaimResponse);
+}
+message MsgCreateClaim {
+  string supplierAddress = 1;
+  bytes smstRootHash = 2;
+  string sessionId = 3;
```
Comment (red-0ne) [inline proto/pocket/supplier/tx.proto:14]: Should we transition to using session_header?
cc @Olshansk

File: proto/pocket/supplier/tx.proto [hunk 2]
```diff
@@ -1,7 +1,18 @@
...
 option go_package = "pocket/x/supplier/types";
 
 // Msg defines the Msg service.
-service Msg {}
\ No newline at end of file
+service Msg {
+  rpc CreateClaim (MsgCreateClaim) returns (MsgCreateClaimResponse);
+}
+message MsgCreateClaim {
+  string supplierAddress = 1;
```
Comment (red-0ne) [inline proto/pocket/supplier/tx.proto:12]: We should decide on whether we use camel or snake case for our protos

**Comment by red-0ne** [comment]
>   ignite scaffold message create-claim SupplierAddress SmstRootHash SessionId SessionNumber:uint InvalidationHeight:int --module supplier

InvalidationHeight doesn't seem to be used

File: proto/pocket/supplier/tx.proto (hunk 1 above)
Comment (bryanchriswhite) [reply proto/pocket/supplier/tx.proto:14]: Yes, as discussed, let's have a common `SessionHeader` protobuf message type defined in the session module which will be common to `RelayMets`, `MsgCreateClaim`, and `MsgSubmitProof`.

File: proto/pocket/supplier/tx.proto (hunk 2 above)
Comment (bryanchriswhite) [reply proto/pocket/supplier/tx.proto:12]: Thanks for catching this, it should be snake case.

**Comment by bryanchriswhite** [comment]
> > ignite scaffold message create-claim SupplierAddress SmstRootHash SessionId SessionNumber:uint InvalidationHeight:int --module supplier
> 
> InvalidationHeight doesn't seem to be used

Nice catch! :+1: I had already removed it from the code but forgot to update the scaffold command in the description.

File: x/supplier/keeper/msg_server_create_claim.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"pocket/x/supplier/types"
+)
+
+func (k msgServer) CreateClaim(goCtx context.Context, msg *types.MsgCreateClaim) (*types.MsgCreateClaimResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// TODO: Handling the message
```
Comment (bryanchriswhite) [inline x/supplier/keeper/msg_server_create_claim.go:13]:
```suggestion
	/*
		INCOMPLETE: Handling the message

		## Validation

		### Session validation
		1. [ ] claimed session ID == retrieved session ID
		2. [ ] this supplier is in the session's suppliers list

		### Msg distribution validation (depends on session validation)
		1. [ ] pseudo-randomize earliest block offset
		2. [ ] governance-based earliest block offset

		### Claim validation
		1. [ ] session validation
		2. [ ] msg distribution validation

		## Persistence
		1. [ ] create claim message
			- supplier address
			- session header
			- claim
		2. [ ] last block height commitment; derives:
			- last block committed hash, must match proof path
			- session ID (?)
	*/
```

File: x/supplier/client/cli/tx_create_claim.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
+package cli
+
```
Comment (Olshansk) [inline x/supplier/client/cli/tx_create_claim.go:2]:
```suggestion
	TODO(@bryanchriswhite): Add unit tests for the CLI command when implementing the business logic.
```

See #59 for an example

File: x/supplier/types/message_create_claim_test.go [hunk 5]
```diff
@@ -0,0 +1,40 @@
+package types
+
```
Comment (Olshansk) [inline x/supplier/types/message_create_claim_test.go:2]:
```suggestion
	TODO(@bryanchriswhite): Add unit tests for message validation when adding the business logic.
```

See #59 for an example

File: x/supplier/keeper/msg_server_create_claim.go [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"pocket/x/supplier/types"
+)
+
+func (k msgServer) CreateClaim(goCtx context.Context, msg *types.MsgCreateClaim) (*types.MsgCreateClaimResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	/*
+		INCOMPLETE: Handling the message
```
Comment (Olshansk) [inline x/supplier/keeper/msg_server_create_claim.go:14]: Great comments!

**Review by Olshansk — APPROVED** [review]

**Review by Olshansk — APPROVED** [review]
LGTM.

@red-0ne I saw your comments but believe it's okay to merge it in without resolving them. I see them as guidelines for the reviewer to document and share what the next step is, but not anything set in stone.

I think it's a great example of why we need to keep these PRs small because we can potentially have a whole day of discussion on each of the bullet points, but need to keep things moving in smaller chunks.

File: x/supplier/keeper/msg_server_create_claim.go [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+		### Claim validation
+		1. [ ] session validation
+		2. [ ] msg distribution validation
+
+		## Persistence
+		1. [ ] create claim message
+			- supplier address
+			- session header
+			- claim
+		2. [ ] last block height commitment; derives:
```
[the remark this answers is not in the document]
Comment (bryanchriswhite) [reply x/supplier/keeper/msg_server_create_claim.go:35]: It is, but it must be committed to already at the  point.

[the remark this answers is not in the document]
Comment (bryanchriswhite) [reply x/supplier/keeper/msg_server_create_claim.go:35]: (Not to be confused with the pseudo-random input for the proof path.) :sweat_smile:

**Pull request merged**